titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/common/V3QueueableTask.java

```diff
@@ -237,4 +299,13 @@ private void setupCustomNamedResources(Job<?> job) {
     private static String getConcatenatedSecurityGroups(List<String> securityGroups) {
         return String.join(":", securityGroups);
     }
+
+    private static int initialOpportunisticCpuCount(JobDescriptor jobDescriptor) {
+        return initialOpportunisticCpuCount(jobDescriptor.getContainer().getContainerResources().getCpu());
+    }
+
+    private static int initialOpportunisticCpuCount(double cpusRequested) {
+        return (int) Math.floor(cpusRequested);
```

Comment:
if we ever have fractional CPU asks, I'm making an assumption that only the floor integer part of it will be allocated opportunistically, and the fractional part would come from regular CPUs

Reply:
This sounds like an architectural decision, with some good rationale behind. Could you write a short comment here why is that?